connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java

```diff
@@ -0,0 +1,469 @@
+ * limitations under the License.
+ */
+
+package com.google.flink.connector.gcp.bigtable.serializers;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.DecimalType;
+import org.apache.flink.table.types.logical.LogicalType;
+
+import com.google.cloud.bigtable.data.v2.models.RowMutationEntry;
+import com.google.flink.connector.gcp.bigtable.testingutils.TestingUtils;
+import com.google.flink.connector.gcp.bigtable.utils.ErrorMessages;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Unit tests for the {@link GenericRecordToRowMutationSerializer} class.
+ *
+ * <p>This class verifies the functionality of the {@link RowDataToRowMutationSerializer} by testing
+ * its ability to serialize {@link RowData} objects into {@link RowMutationEntry} objects, with and
+ * without nested fields. It also includes tests for data type conversions and error handling.
+ */
+public class RowDataTest {
+    private final DataType dtSchema =
+            DataTypes.ROW(
+                    DataTypes.FIELD(TestingUtils.ROW_KEY_FIELD, DataTypes.STRING()),
+                    DataTypes.FIELD(TestingUtils.STRING_FIELD, DataTypes.STRING()),
+                    DataTypes.FIELD(TestingUtils.INTEGER_FIELD, DataTypes.INT()));
+
+    private final DataType dtNestedSchema =
+            DataTypes.ROW(
+                    DataTypes.FIELD(TestingUtils.ROW_KEY_FIELD, DataTypes.STRING()),
+                    DataTypes.FIELD(
+                            TestingUtils.NESTED_COLUMN_FAMILY_1,
+                            DataTypes.ROW(
+                                    DataTypes.FIELD(TestingUtils.STRING_FIELD, DataTypes.STRING()),
+                                    DataTypes.FIELD(TestingUtils.INTEGER_FIELD, DataTypes.INT()))),
+                    DataTypes.FIELD(
+                            TestingUtils.NESTED_COLUMN_FAMILY_2,
+                            DataTypes.ROW(
+                                    DataTypes.FIELD(
+                                            TestingUtils.STRING_FIELD_2, DataTypes.STRING()))));
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testCorrectSerializerInitialization(Boolean useNestedRowsMode) {
+        RowDataToRowMutationSerializer.Builder builder =
```

Comment:
can we also add a test for null values on all the initialization configuration fields?

Reply:
Added two test for null key and null schema.  I did the same for GenericRecord